platform/mbed_mem_trace.cpp

```diff
@@ -45,6 +46,24 @@ void mbed_mem_trace_set_callback(mbed_mem_trace_cb_t cb) {
     mem_trace_cb = cb;
 }
 
+void mbed_mem_trace_disable()
+{
+    mbed_mem_trace_lock();
+    if (mem_trace_cb) {
+        mem_trace_cb_reserve = mem_trace_cb;
+        mem_trace_cb = 0;
+    }
+    mbed_mem_trace_unlock();
+}
+void mbed_mem_trace_enable()
+{
+    mbed_mem_trace_lock();
+    if (mem_trace_cb == 0 && mem_trace_cb_reserve) {
```

Comment:
'If (mem_trace_cb == 0)' this is inconsistent with the previous function where 'if (mem_trace_cb)' is used. One is checking a value directly and one is treating the variable as a logical True/False (which I assume is how it should be used). If the latter is indeed true then the checking against 0 should actually be replaced by:
if (!mem_trace_cb).